main/src/ca/uwaterloo/flix/language/phase/Safety.scala

```diff
@@ -270,14 +284,46 @@ object Safety extends Phase[Root, Root] {
       case Polarity.Negative =>
         // Compute the free variables in the terms which are *not* bound by the lexical scope.
         val freeVars = terms.flatMap(freeVarsOf).toSet intersect quantVars
+        val errors = checkNegativeWildcards(terms, loc)
 
         // Check if any free variables are not positively bound.
-        ((freeVars -- posVars) map {
-          case unboundVar => SafetyError.IllegalNonPositivelyBoundVariable(unboundVar, loc)
-        }).toList
+        errors ++ ((freeVars -- posVars) map (unboundVar => SafetyError.makeIllegalNonPositivelyBoundVariableError(unboundVar, loc))).toList
```

Comment:
This is not easy to read.

I would make two local variables with meaningful names and then you can ++ them at the end.